### PR TITLE
Use new Surface API

### PR DESCRIPTION
**Pull request by beaufortfrancois**
On Emscripten, it requires https://github.com/emscripten-core/emscripten/pull/21939/

### CODE REVIEW

File: main.cpp
Comment (Kangz) [inline main.cpp:58]: Note that this will be part of the descriptor soon.

**Review by Kangz — APPROVED** [review]

File: main.cpp
Comment (kainino0x) [reply main.cpp:56]: (It is already there in Dawn, but Emscripten doesn't implement it.)

File: main.cpp
Comment (kainino0x) [inline main.cpp:41]: nit: print the `message`

File: main.cpp
Comment (kainino0x) [inline main.cpp:54]: nit: print the `message` if there's an error (though this shouldn't ever happen because it only happens when you make a malformed request)

File: main.cpp
Comment (kainino0x) [inline main.cpp:27]: Use the preferred format instead of format[0], I think, though I have filed https://github.com/webgpu-native/webgpu-headers/issues/290 about the fact that this is not clear.

Comment (beaufortfrancois) [reply main.cpp:27]: I'll be happy to replace this line with `.format = surface.GetPreferredFormat(adapter),` but I wonder if https://github.com/webgpu-native/webgpu-headers/issues/290 will get resolved soon and maybe we'll end up using `formats[0]`.

File: main.cpp
Comment (beaufortfrancois) [reply main.cpp:54]: Ditto

File: main.cpp
Comment (kainino0x) [reply main.cpp:41]: I think it makes it a bit more useful as a reference, because if someone tries to run the sample and it fails, they'll see some explanation of why. But no strong opinion.

**Review by Kangz — APPROVED** [review]

File: main.cpp
Comment (kainino0x) [inline main.cpp:28]: Oops, I had a draft suggestion that I never submitted.

```suggestion
      .alphaMode = wgpu::CompositeAlphaMode::Auto,
```

File: main.cpp
Comment (kainino0x) [inline main.cpp:31]:
```suggestion
      .presentMode = wgpu::PresentMode::Fifo,
```

**Review by kainino0x — APPROVED** [review]
LGTM once emscripten changes land

**Comment by beaufortfrancois** [comment]
@kainino0x I've addressed your latest feedback. I'll merge when emscripten changes land. Thank you!

**Review by beaufortfrancois — COMMENTED** [review]
Note that I'm now using default values for alphaMode and presentMode

**Comment by beaufortfrancois** [comment]
I'll merge when Chrome 126 hits stable. Thanks for the reviews!

**Pull request merged**
